Extensions/Leaderboards/tests/leaderboardstools.spec.js

```diff
@@ -28,29 +28,6 @@ describe('Leaderboards', () => {
       ).to.be('aPlayerNameTh4tIsT00LongToBeSa');
     });
 
-    it('it generates a predefined player name with a random number if input is void/wrong type/empty', () => {
```

Comment:
Why remove those tests since the expression `FormatPlayerName` is still present?

Reply:
Good call, added it back, but checking the result is an empty string 👍

Reply:
The problem is that since it's an expression, we're creating a regression: some users might use it in an expression field

Reply:
The expression is described as "Formats a name so that it can be submitted to a leaderboard", so unsure if we should still support something that formats locally with PlayerXxxx... as this could mean someone will on the contrary be confused that this does not honor the configuration of the leaderboard (but the expression has no way to know this)

Reply:
I think this can still be used then! You use the expression and either get:

- A properly formatted player name.
- Or if you passed something empty, well it's still empty but now the backend accepts it (so the end result should not be different).

I'm willing to try this with a "breaking change" indicating that if a player name is empty, it will be left empty but that for most games it should be ok (the player probably don't care what their name is if they don't enter anything). If there is an actual "regression" reported, we either see if it's a matter of asking to adapt the game, or if we must revert the change to the expression.